src/Core/functions.php

```diff
@@ -25,13 +25,14 @@
     /**
      * Convenience method for htmlspecialchars.
      *
-     * @param string|array|object $text Text to wrap through htmlspecialchars. Also works with arrays, and objects.
+     * @param mixed $text Text to wrap through htmlspecialchars. Also works with arrays, and objects.
```

Comment:
We shoudn't weaken it maybe? It should only be a string or array usually, it seems it can also be an object though.

Reply:
I just fixed the docblock. You can already pass practically any data type to the method.

Reply:
That is always true in php 5 classes. We still only typehint expected types.

Reply:
PHP version doesn't matter. Even in 4.x we won't be able to add any actual type hint for `$text` argument since it can be of multiple type. So replacing `mixed` with list of half a dozen or so types would be silly.

Reply:
I am a bit worried about this method losing documention on its primary intent: `string|string[]` - it works on strings, nothing else. The others can be passed for simplicity only, and for performance make a no-op. But passing those is actually a bit silly and as I mentioned for simplicity of the templates.
So while yours functionally correct in the strict sense, the doc block here can still state more the intend of the method IMO and the expected primary input types (rest is just noise).